pkg/controller/controller.go

```diff
@@ -143,14 +145,35 @@ func (c *Controller) handlePopupPost(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// 3. [TODO] Request a token
-	token := "token_from_server"
+	// 3. Request a token
+	ttl, err := strconv.Atoi(formDetails.TTL)
```

Comment:
We should accept ttls as strings (e.g. "12s") and use `time.ParseDuration`.

Reply:
cool didnt know about this library, this simplifies things.